### PR TITLE
Local event buffer

### PR DESCRIPTION
**Pull request by sebn**

### CODE REVIEW

File: cli/src/local/watcher.js [hunk 1]
```diff
@@ -125,6 +137,26 @@ class LocalWatcher {
...
+          break
+        case 'addDir':
+          this.onAddDir(e.path, e.stats)
+          break
+        case 'change':
+          this.onChange(e.path, e.stats)
+          break
+      }
+    })
+    _.each(unlinkDirEvents, e => { this.onUnlinkDir(e.path) })
```
Comment (sebn) [inline cli/src/local/watcher.js]: @aenario Tests pass with this version, but I'm almost sure it's actually broken.
Here we naively change the order of the events, which may fix some event order issues but probably also breaks other cases...

Passing the events to the LocalWatcher methods as they come from the buffer would probably match the previous behavior more closely, but it breaks the tests.

**Comment by sebn** [comment]
@aenario Code looks good to me (WDYT?).

I'll do a bit more manual testing on both platforms and we should be good to go 😄

**Comment by sebn** [comment]
- [ ] Logging should be updated too, I'll fix it.

File: cli/src/local/watcher.js [hunk 2]
```diff
@@ -125,11 +135,37 @@ class LocalWatcher {
...
+      }
+    }
+  }
+
   stop () {
     if (this.watcher) {
       this.watcher.close()
       this.watcher = null
     }
+    this.buffer.switchMode('idle')
```
Comment (aenario) [inline cli/src/local/watcher.js]: Shouldn't we flush there ?

Comment (sebn) [reply cli/src/local/watcher.js]: The buffer flushes automatically when switching mode, because in both cases we want to flush anyway.

**Comment by sebn** [comment]
- [ ] Not done with manual testing yet.

**Pull request merged**